internal/runner/runner.go

```diff
@@ -97,12 +110,13 @@ func (r *Runner) Rollback(hr v2.HelmRelease) error {
 	rollback.Force = hr.Spec.GetRollback().Force
 	rollback.Recreate = hr.Spec.GetRollback().Recreate
 	rollback.CleanupOnFail = hr.Spec.GetRollback().CleanupOnFail
+	rollback.Version = hr.Status.LastSuccessfulReleaseRevision
```

Comment:
> There is an existing release, such as one previously managed by the Helm Operator, and Helm controller takes it over.

If `rollback.Version` is set to `0`, Helm will roll it back to the previous revision.

> User manually fixes a helm release, perhaps via a suspend/resume flow.

Given this creates "revision drift", it would always trigger a new release and I would expect this to succeed because of the fixes performed by the user (and later reflected to the chart and/or `HelmRelease` before resuming).

> Also, if a previous reconciliation had a failed upgrade which was not rolled back, then do we really want to rollback to a previous release which may have been made a long time ago and may be out of date?

I am hesitant about this because of the following:

1. We have little knowledge what the contents of the previous revision are, or if it has been tampered with.
2. We have little knowledge about the state of previous revision before the current revision was created (was it e.g. `StatusFailed`?), as any "previous" revision gets a `StatusSuperseded` from Helm.
3. Our "revision bookkeeping" model would make less sense, as we would detect mutations but allow rollbacks to the previous state if an upgrade on top of this detection fails. Given we then mark the revision of the rolled back release as the new revision, the controller basically would verify state it did not create itself.